src/main/java/net/greatstart/configs/SpringSecurityDev.java

```diff
@@ -0,0 +1,77 @@
+package net.greatstart.configs;
+
+import net.greatstart.services.GreatStartUserDetailsService;
+import net.greatstart.services.UserService;
+import org.h2.server.web.WebServlet;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.web.servlet.ServletRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class SpringSecurityDev extends WebSecurityConfigurerAdapter {
+    private UserService userService;
+
+    @Autowired
+    public SpringSecurityDev(UserService userService) {
+        this.userService = userService;
+    }
+
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth
+                .userDetailsService(new GreatStartUserDetailsService(userService))
+                .passwordEncoder(passwordEncoder());
+    }
+
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .httpBasic().and()
+                .authorizeRequests()
+                .antMatchers("/project/new",
+                        "/project/my",
+                        "/invinterest/add",
+                        "/invinterest",
+                        "/views/main/Help.html",
+                        "/views/user/UserPage.html",
+                        "/views/user/EditUser.html").authenticated()
+                .and()
+                .authorizeRequests().antMatchers("/console/**").permitAll()
+                .and()
+                .formLogin()
+                .loginPage("/user/login")
+                .usernameParameter("email")
+                // TODO: change the routing when home page is ready
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)